stingray/crossspectrum.py

```diff
@@ -1091,6 +1282,164 @@ def classical_significances(self, threshold=1, trial_correction=False):
 
         return pvals
 
+    @staticmethod
+    def from_time_array(*args, **kwargs):
+        """Calculate AveragedCrossspectrum from two arrays of event times.
+
+        Parameters
+        ----------
+        times1 : `np.array`
+            Event arrival times of channel 1
+        times2 : `np.array`
+            Event arrival times of channel 2
+        dt : float
+            The time resolution of the intermediate light curves
+            (sets the Nyquist frequency)
+
+        Other parameters
+        ----------------
+        segment_size : float
+            The length, in seconds, of the light curve segments that will be averaged.
+            Only relevant for AveragedCrossspectrum
+        gti : [[gti0, gti1], ...]
+            Common Good Time intervals
+        norm : str, default "abs"
+            The normalization of the periodogram. "abs" is absolute rms, "frac" is
+            fractional rms, "leahy" is Leahy+83 normalization, and "none" is the
+            unnormalized periodogram
+        use_common_mean : bool, default True
+            The mean of the light curve can be estimated in each interval, or on
+            the full light curve. This gives different results (Alston+2013).
+            Here we assume the mean is calculated on the full light curve, but
+            the user can set ``use_common_mean`` to False to calculate it on a
+            per-segment basis.
+        fullspec : bool, default False
+            Return the full periodogram, including negative frequencies
+        silent : bool, default False
+            Silence the progress bars
+        power_type : str, default 'all'
+            If 'all', give complex powers. If 'abs', the absolute value; if 'real',
+            the real part
+        """
+
+        return crossspectrum_from_time_array(*args, **kwargs)
+
+    @staticmethod
+    def from_events(*args, **kwargs):
+        """Calculate AveragedCrossspectrum from two event lists
```

Comment:
I wrote these methods to initialize the crossspectrum from a single kind of input. I know it seems kind of contradictory to what I've done somewhere else, but in these methods I like being extra clear. from_events ? Then the input is an event list. from_lightcurve? light curves. etc. We might include the numpy array inside from_events, but then we would need to specify in which cases gtis have to be input by hand or get ignored... Feasible, but I think it loses some clarity.

Reply:
Eventually, I would like the users to always use `cs = Crossspectrum.from_events(events1, events2)` when they have event lists.
I kept the current interface because doing otherwise would be a major breaking change, but I think that in the future we should be initializing Crossspectrum objects with the `__init__` method by passing `freq`, `power`, while using specialized static methods to make calculations that create Crossspectra.